frontend/src/metabase/query_builder/components/notebook/lib/steps.ts

```diff
@@ -142,13 +145,21 @@ export function getQuestionSteps(question, openSteps = {}) {
 /**
  * Returns an array of "steps" to be displayed in the notebook for one "stage" (nesting) of a query
  */
-export function getStageSteps(stageQuery, stageIndex, openSteps) {
-  const getId = (step, itemIndex) =>
-    `${stageIndex}:${step.type}` + (itemIndex > 0 ? `:${itemIndex}` : ``);
-
-  function getStep(STEP, itemIndex = null) {
+export function getStageSteps(
+  stageQuery: StructuredQuery,
+  stageIndex: number,
+  openSteps: Record<NotebookStep["id"], boolean>,
```

Comment:
This type looks the same as the one defined in `NotebookSteps` on line 19. Is it possible to refactor and use the same type in both places?

Reply:
Good point! Let me tweak it in #29555